portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,12 +38,24 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comment");
     PreparedQuery results = datastore.prepare(query);
- 
+
+    int numComments = getNumComments(request);
+    if (numComments == -1) {
```

Comment:
`numComments < 0` is better practice. It will catch any negative input value (say I request -3?).